server/server.go

```diff
@@ -808,6 +809,15 @@ func (s *Server) SetReplicationConfig(cfg config.ReplicationConfig) error {
 			}
 		}
 	}
+	if cfg.EnablePlacementRules {
```

Comment:
I think we can move it to validate and no need to compare it with the old config.

Reply:
It looks we cannot know if `MaxReplicas` or `LocationLables` is changed without comparing.

Reply:
Oh, right. But I still think this validate logic should be put into the validate funciton.

Reply:
But the config is valid, only the config transition is invalid.

Reply:
make sense